app/controllers/application_controller.rb

```diff
@@ -3,4 +3,15 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   include SessionsHelper
+
+  private
+
+  def query_header_for_query(query, any_search_results)
```

Comment:
Since this method is only being used in the controllers, we don't need to worry about it being available in the views. I was thinking we could do something like we've done with `SessionsHelper`, but instead of including the new helper here in `ApplicationController`, we could include it where it's being used (`ItemsController` and `ArtistsController`). We don't have to make this change, it was just a thought.

Concerns are just another way to share reusable code. I think the most we could gain by using one here is some code organization. Concerns are more useful when there is a lot of stuff that needs to be shared. You can see it being used in models here: http://richonrails.com/articles/rails-4-code-concerns-in-active-record-models.